gcloud/datastore/connection.py

```diff
@@ -68,13 +70,16 @@ def _request(self, dataset_id, method, data):
             uri=self.build_api_url(dataset_id=dataset_id, method=method),
             method='POST', headers=headers, body=data)
 
-        if headers['status'] != '200':
-            raise Exception('Request failed. Error was: %s' % content)
+        status = headers['status']
+        if status != '200':
+            message = ('Request failed with status code %s. '
+                       'Error was: %s' % (status, content))
+            raise six.moves.http_client.HTTPException(message)
```

Comment:
Rather than return the base `HTTPException`, we should probably be mapping the status code onto the appropriate subclass (maybe promoting `gcloud.storage.exceptions` up to `gcloud.exceptions`?)